### PR TITLE
feat: add support for kuma service mesh

### PR DESCRIPTION
**Pull request by bartsmykla**
- Updated make files to build helm chart and k8s deployment
  manifest with Kuma support
- Bumped helm chart to 0.5.0
- Kuma supports DNS redirection like Istio does, so I updated
  ebpf files which so far were only used for Istio, to also
  be compiled for Kuma
- As Kuma also supports dynamic reload, I updated hack/reload-prog.sh
  to be able to use it with Kuma
   - shebang has to be in the first line of the script, so I moved
     it there
   - as the shell code is definitelly written with bash in mind
     (i.e. usage of arrays), I changed shebang from `sh` to `bash`
- Updated all README.md files with kuma-related instructions and
  informations
- Added `scripts/install-kuma.sh` script which can be used
  to install kuma and wait till the deployment will be ready
- Modified github actions workflow to include e2e test for kuma

Signed-off-by: Bart Smykla <bartek@smykla.com>

### CODE REVIEW

**Comment by kebe7jun** [comment]
Almost LGTM.
Just need to take care of the errors in the style check and this PR will be merged! Thank you for your approval of the Merbridge project!

**Comment by kebe7jun** [comment]
https://github.com/merbridge/merbridge/runs/7176529801?check_suite_focus=true#step:8:26

Should the operation wait until the kuma service starts?

**Review by Xunzhuo — DISMISSED** [review]
LGTM thanks! @bartsmykla

**Comment by bartsmykla** [comment]
I see some kuma e2e tests are failing, I'm looking into it right now

**Comment by Xunzhuo** [comment]
It looks like e2e tests for kuma failed for some reasons, could you take a look at this @bartsmykla thanks!

**Comment by Xunzhuo** [comment]
Thank you @bartsmykla, not in a hurry, just take a look when you are free.

**Review by Xunzhuo — APPROVED** [review]

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -75,3 +75,166 @@ jobs:
...
+                image: curlimages/curl
+                command: ["/bin/sleep", "infinity"]
+                imagePullPolicy: IfNotPresent' | kubectl apply -f -
+        
+        # wait till our example deployment are fully ready
+        kubectl rollout status deployment example-server
+        kubectl rollout status deployment example-client
+    - name: test connect without Merbridge
+      run: |
+        sleep 5
```
Comment (kebe7jun) [inline .github/workflows/e2e.yaml:217]: It is recommended not to use sleep as a condition for Pod ready, it is better to judge whether the Pod status is running.

**Review by kebe7jun — COMMENTED** [review]
jsut some nits.

File: .github/workflows/e2e.yaml (hunk 1 above)
Comment (bartsmykla) [reply .github/workflows/e2e.yaml:217]: Ack. I'm gonna fix it

Comment (kebe7jun) [reply .github/workflows/e2e.yaml]: e2e test still failed.  😂

If sleep is ok, we can keep this.

Comment (bartsmykla) [reply .github/workflows/e2e.yaml]: ok, finally done!

**Review by kebe7jun — APPROVED** [review]
Thank you very much!

**Pull request merged**